From 01e7646678836ab83f540b2f6fc58ad1992efa54 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Mon, 28 May 2018 15:06:25 +0800
Subject: [PATCH 607/666] driver: freescale: sdk_dpaa: fix incompatible pointer
 type

Avoid below building error:

drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_generic.c:87:21: error:
initialization from incompatible pointer type
[-Werror=incompatible-pointer-types]
|   .ndo_get_stats64 = dpa_generic_get_stats64,
	|                      ^~~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c  | 4 +---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.h  | 2 +-
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_generic.c | 6 ++----
 3 files changed, 4 insertions(+), 8 deletions(-)
 mode change 100644 => 100755 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c
 mode change 100644 => 100755 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.h
 mode change 100644 => 100755 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_generic.c

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c
old mode 100644
new mode 100755
index 9385d51..b2112b0
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.c
@@ -253,7 +253,7 @@ EXPORT_SYMBOL(dpa_timeout);
  * Calculates the statistics for the given device by adding the statistics
  * collected by each CPU.
  */
-struct rtnl_link_stats64 * __cold
+void __cold
 dpa_get_stats64(struct net_device *net_dev,
 		struct rtnl_link_stats64 *stats)
 {
@@ -272,8 +272,6 @@ dpa_get_stats64(struct net_device *net_dev,
 		for (j = 0; j < numstats; j++)
 			netstats[j] += cpustats[j];
 	}
-
-	return stats;
 }
 EXPORT_SYMBOL(dpa_get_stats64);
 
diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.h b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.h
old mode 100644
new mode 100755
index 4706e90..1361906
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.h
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_common.h
@@ -144,7 +144,7 @@ int dpa_netdev_init(struct net_device *net_dev,
 int __cold dpa_start(struct net_device *net_dev);
 int __cold dpa_stop(struct net_device *net_dev);
 void __cold dpa_timeout(struct net_device *net_dev);
-struct rtnl_link_stats64 * __cold
+void __cold
 dpa_get_stats64(struct net_device *net_dev,
 		struct rtnl_link_stats64 *stats);
 int dpa_change_mtu(struct net_device *net_dev, int new_mtu);
diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_generic.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_generic.c
old mode 100644
new mode 100755
index 1fdde61..579282d
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_generic.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_generic.c
@@ -61,7 +61,7 @@ static uint16_t tx_timeout = 1000;
 module_param(tx_timeout, ushort, S_IRUGO);
 MODULE_PARM_DESC(tx_timeout, "The Tx timeout in ms");
 
-struct rtnl_link_stats64 *__cold
+void __cold
 dpa_generic_get_stats64(struct net_device *netdev,
 			struct rtnl_link_stats64 *stats);
 static int dpa_generic_set_mac_address(struct net_device *net_dev,
@@ -103,7 +103,7 @@ static void dpa_generic_draining_timer(unsigned long arg)
 		mod_timer(&(priv->timer), jiffies + 1);
 }
 
-struct rtnl_link_stats64 *__cold
+void __cold
 dpa_generic_get_stats64(struct net_device *netdev,
 			struct rtnl_link_stats64 *stats)
 {
@@ -122,8 +122,6 @@ dpa_generic_get_stats64(struct net_device *netdev,
 		for (j = 0; j < numstats; j++)
 			netstats[j] += cpustats[j];
 	}
-
-	return stats;
 }
 
 static int dpa_generic_set_mac_address(struct net_device *net_dev,
-- 
2.7.4

